addons/hr_holidays/models/hr_leave.py

```diff
@@ -777,8 +777,11 @@ def onchange(self, values, field_name, field_onchange):
         # Try to force the leave_type name_get when creating new records
         # This is called right after pressing create and returns the name_get for
         # most fields in the view.
-        if 'employee_id' in field_onchange:
-            self = self.with_context(employee_id=int(field_onchange['employee_id']))
+        if 'employee_id' in field_onchange and 'employee_id' not in self._context:
+            employee_ids_list = [value[2] for value in values.get('employee_ids', []) if len(value) == 3 and value[0] == Command.SET]
+            employee_ids = employee_ids_list[-1] if employee_ids_list else []
+            employee_id_value = employee_ids[0] if employee_ids else False
+            self = self.with_context(employee_id=employee_id_value or self._context.get('default_employee_id', self.env.user.employee_id.id))
```

Comment:
As this logic is the same as in `hr_leave_allocation`, it would be better to define it as a top-level function (like `get_employee_id_from_context`) here and then import it over there :-)

Reply:
Done, thanks :-)